### ISSUE TITLE
[stable/postgresql] statefulset-slaves mount secrets bug

### ISSUE THREAD

**Issue by mruepp** [helm/charts#13652]
**Describe the bug**
Bug in the volumes and volumeMount section of statefulset-slaves.yaml
The following formatting is buggy:
```
+++ b/stable/postgresql/templates/statefulset-slaves.yaml
@@ -166,8 +166,8 @@ spec:
         volumeMounts:
         {{- if .Values.usePasswordFile }}
         - name: postgresql-password
-          mountPath: /opt/bitnami/postgresql/secrets
-        {{ end }}
+          mountPath: /opt/bitnami/postgresql/secrets/
+        {{- end }}
         {{- if .Values.persistence.enabled }}
         - name: data
           mountPath: {{ .Values.persistence.mountPath }}
@@ -186,7 +186,7 @@ spec:
       - name: postgresql-password
         secret:
           secretName: {{ template "postgresql.secretName" . }}
-      {{ end }}
+      {{- end }}
       {{- if or (.Files.Glob "files/postgresql.conf") (.Files.Glob "files/pg_hba.conf") .Values.postgresqlConfiguration .Values.pgHbaConfiguration .Values.configurationConfigMap}}
       - name: postgresql-config
         configMap:
```

**Version of Helm and Kubernetes**:
helm 2.12.3
kubernetes 1.13.5

**Which chart**:
stable/postgresql

**What happened**:
On the slave pod initialization, the log throws:
```
ERROR ==> The POSTGRES_PASSWORD environment variable is empty or not set. Set the environment variable ALLOW_EMPTY_PASSWORD=yes to allow the container to be started with blank passwords. This is recommended only for development.
```
The Secret key "postgresql-password" don't get mounted in the slave pods

This is the case with using secrets OR using file by setting 
``` 
usePasswordFile: true
```
in the values.yaml and values-production.yaml

**What you expected to happen**:
slave pods should initialize properly, the secret key
"postgresql-password" should get mounted into:
mountPath: /opt/bitnami/postgresql/secrets/

**How to reproduce it** (as minimally and precisely as possible):
Deploy stable/postgres from scratch

**Anything else we need to know**:

**Comment by juan131** [comment]
Hi @mruepp 

Did you check the section below in the README.md?

- https://github.com/helm/charts/tree/master/stable/postgresql#400

The env. variable POSTGRES_PASSWORD is mandatory now and you should try to upgrade your PostgreSQL chart to the version **4.0.0**

**Comment by mruepp** [comment]
FYI @tompizmor @desaintmartin @juan131

**Comment by mruepp** [comment]
@juan131 Hi, we use the master chart

**Comment by juan131** [comment]
Hi @mrueg 

What image version are you using? Is it **10.7.0-r69**?

**Comment by mruepp** [comment]
Image as defined in the values-production.yaml: 10.7.0
But the secrets mounting issue is resolved if refactoring the following lines:
```
         volumeMounts:
         {{- if .Values.usePasswordFile }}
         - name: postgresql-password
-          mountPath: /opt/bitnami/postgresql/secrets
-        {{ end }}
+          mountPath: /opt/bitnami/postgresql/secrets/
+        {{- end }}
```
and
```
       - name: postgresql-password
         secret:
           secretName: {{ template "postgresql.secretName" . }}
-      {{ end }}
+      {{- end }}
```
in templates/statefulset-slaves.yaml

There is a missing slash at the end of mountPath: /opt/bitnami/postgresql/secrets so the secret key or file don't get mounted properly

**Comment by juan131** [comment]
Thanks for reporting that @mruepp 

I just created a PR to address this issue.

**Comment by mruepp** [comment]
In the statefulset.yaml (master) the snippet is different and works.
```
{{- if .Values.usePasswordFile }}
        - name: postgresql-password
          mountPath: /opt/bitnami/postgresql/secrets/
        {{- end }}
```
```
volumes:
...
      {{- if .Values.usePasswordFile }}
      - name: postgresql-password
        secret:
          secretName: {{ template "postgresql.secretName" . }}
      {{- end }}
```

**Comment by juan131** [comment]
Hi @mruepp 

The PR has just been merged and the new version **4.0.1** address this specific issue. Thanks again for reporting it.
